lib/Resource.js

```diff
@@ -104,7 +115,8 @@ Resource.prototype._associateWith = function(resource) {
     }).pop();
     if (!match) return;
 
-    if (relation.meta.many) {
+    if (Resource._rawRelationHasMany(relation)) {
```

Comment:
We sure can. I would prefer to rename here though, because we're using `relation` to refer to both the related resources collection and the relationship metadata.